awx/main/tasks/receptor.py

```diff
@@ -266,139 +267,119 @@ def __init__(self, task, runner_params=None):
             self.runner_params['only_transmit_kwargs'] = True
 
     def run(self):
-        # We establish a connection to the Receptor socket
-        receptor_ctl = get_receptor_ctl()
-
         res = None
+        unit_id = None
         try:
-            res = self._run_internal(receptor_ctl)
+            res, unit_id = self._run()
+
+            if res.status == 'error':
+                self.handle_error(unit_id)
+
             return res
         finally:
             # Make sure to always release the work unit if we established it
-            if self.unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
+            if unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
                 try:
-                    receptor_ctl.simple_command(f"work release {self.unit_id}")
+                    self.receptor_ctl.simple_command(f"work release {unit_id}")
```

Comment:
If the concept is the, generally, receptor commands are not safe from hangs, then I expect that would apply to `receptorctl work release` as well. I believe I've had it hang here before.